frontend/src/pages/Notifications/helper.js

```diff
@@ -1,29 +1,91 @@
 import strings from "../../localizeStrings";
 import { formatString } from "../../helper";
 
-// get hierarchic deepest displayName
-export function getDisplayName(notification) {
+const project = Symbol("Project");
+const subproject = Symbol("Subproject");
+const workflowitem = Symbol("Workflowitem");
+
+export function getParentData(notification) {
+  const redacted = "Redacted";
   const metadata = notification.metadata;
+  let projectDisplayName = "";
+  let subprojectDisplayName = "";
+
   if (metadata !== undefined) {
-    if (metadata.workflowitem !== undefined && metadata.workflowitem.hasViewPermissions === true) {
-      return metadata.workflowitem.displayName;
-    }
-    if (metadata.subproject !== undefined && metadata.subproject.hasViewPermissions === true) {
-      return metadata.subproject.displayName;
+    if (metadata.project) {
+      projectDisplayName = isAllowedToSeeType(metadata, project) ? getDisplayName(metadata, project) : redacted;
+      if (metadata.subproject) {
+        subprojectDisplayName = isAllowedToSeeType(metadata, subproject)
+          ? getDisplayName(metadata, subproject)
+          : redacted;
+      }
     }
-    if (metadata.project !== undefined && metadata.project.hasViewPermissions === true) {
-      return metadata.project.displayName;
+  }
+
+  return {
+    projectDisplayName,
+    subprojectDisplayName
+  };
+}
+
+export const getType = notificationMetaData => {
+  if (notificationMetaData.project !== undefined) {
+    if (notificationMetaData.subproject !== undefined) {
+      if (notificationMetaData.workflowitem !== undefined) {
+        return workflowitem;
+      }
+      return subproject;
     }
+    return project;
+  }
+  return null;
+};
+
+export function getDisplayName(notificationMetaData, type) {
+  if (type === project) {
+    return notificationMetaData.project.displayName;
+  }
+  if (type === subproject) {
+    return notificationMetaData.subproject.displayName;
+  }
+  if (type === workflowitem) {
+    return notificationMetaData.workflowitem.displayName;
   }
   return "";
 }
 
+const isAllowedToSeeType = (notificationMetaData, type) => {
+  if (type === project) {
```

Comment:
Make uppercase

Reply:
Use switch case statement

Reply:
changed to uppercase and used switch statement